src/main/java/org/folio/edge/sip2/repositories/ConfigurationRepository.java

```diff
@@ -51,102 +59,135 @@ public ConfigurationRepository(IResourceProvider<IRequestData> resourceProvider,
    */
   public Future<ACSStatus> getACSStatus(SessionData sessionData) {
 
-    ACSStatus.ACSStatusBuilder builder = ACSStatus.builder();
+    LinkedHashMap<String, String> tenantLevelQueryParams = new LinkedHashMap<>();
```

Comment:
Yes, LinkedHashmap is required. Consider the output that we want: 
`param1==value1 OR param2==value2 AND param3==value3`

Surely we can't mix param2 or param3.  Having LinkedHashmap also benefits unit testing because we can do assertions on the whole string as expected, and not having to pick apart the string in order to assert on its parts.

Reply:
Maybe I am misunderstanding the code then, but it looks like we pass a List of Maps to the query arg builder and all key/values in one map are joined between parens with ANDs and each of these is joined by an OR. What do the individual order of the params in each map have to do with that? I can see what you mean for the unit test case though.

Either way, it's not a big deal. I just find the code a bit complicated and want to see if there are ways to simplify it. Typically over-specifying the collection type when declared is frowned upon, which piqued my interest here.